### PR TITLE
Replace anies with types

### PR DESCRIPTION
**Pull request by adam-stover**
Adds some types to take care of the check code complaints.

### CODE REVIEW

File: app/navigation/stack-param-lists.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export type RootStackParamList = {
   authenticationCheck: undefined
   authentication: {
     screenPurpose: AuthenticationScreenPurpose
-    isPinEnabled: boolean
+    isPinEnabled: boolean | undefined
```
Comment (nicolasburtey) [inline app/navigation/stack-param-lists.ts:19]: doesn't feel right for this to be undefined.

File: app/navigation/stack-param-lists.ts [hunk 2]
```diff
@@ -86,10 +86,11 @@ export type MoveMoneyStackParamList = {
 }
 
 export type PhoneValidationStackParamList = {
+  Primary: undefined
```
Comment (nicolasburtey) [inline app/navigation/stack-param-lists.ts:89]: does it means `Primary` can only been `undefined` here? is `Primary` needed then? same question for `welcomePhoneInput` below I guess

Comment (adam-stover) [reply app/navigation/stack-param-lists.ts:89]: These types are used with `react-navigation` to define the routes and any params these routes expect -- setting up `Primary: undefined` allows us to do, for example, `navigator.navigate("Primary")` and TS will auto-suggest `"Primary"`. See https://reactnavigation.org/docs/typescript#type-checking-the-navigator

File: app/navigation/stack-param-lists.ts (hunk 1 above)
Comment (adam-stover) [reply app/navigation/stack-param-lists.ts:19]: yes this is unnecessary, good catch -- the `PhoneValidationStackParamList['authentication']` has the definition we need (to use in `phone-auth.tsx`).

**Review by daviroo — APPROVED** [review]
Reviewed on a call with @adam-stover

**Pull request merged**